migrations/0001_initial_schema.up.sql

```diff
@@ -0,0 +1,50 @@
+create extension if not exists "uuid-ossp";
+
+create table promotions (
+  id uuid primary key not null default uuid_generate_v4(),
+  promotion_type text not null,
+  created_at timestamp with time zone not null default current_timestamp,
+  expires_at timestamp with time zone not null default current_timestamp + interval '4 months',
+  version integer not null default 5,
+  suggestions_per_grant integer not null,
+  approximate_value numeric(28, 18) not null check (approximate_value > 0.0),
+  remaining_grants integer not null check (remaining_grants >= 0),
+  active boolean not null default false
+);
+
+alter table promotions add constraint check_promotion_type check (promotion_type in ('ugp', 'ads'));
+
+create table issuers (
+  promotion_id uuid not null references promotions(id),
+  cohort text not null,
+  public_key text not null,
+  primary key (promotion_id, cohort)
+);
+
+create table wallets (
+  id uuid primary key not null,
+  -- created_at timestamp with time zone not null default current_timestamp,
```

Comment:
do we not want this info?

Reply:
it's not super useful unless we're synchronizing with the ledger held value. let me think on it more - maybe we should do that before moving forward